integrations/ninja-halo-guid-sync.ps1

```
@@ -0,0 +1,245 @@
+$NinjaBaseUrl      = "https://app.ninjarmm.com"
+
+# --- AUTHENTICATE: HALO ---
+$haloTokenResp = Invoke-RestMethod -Method Post -Uri "$HaloBaseUrl/auth/token?tenant=$HaloTenant" -ContentType "application/x-www-form-urlencoded" -Body @{
+    grant_type    = "client_credentials"
+    client_id     = $HaloClientId
+    client_secret = $HaloClientSecret
+    scope         = $HaloScope
+}
+$haloToken = $haloTokenResp.access_token
+if (-not $haloToken) {
+    Write-Host "❌ Halo auth failed — response was:"
```

PSScriptAnalyzer (1.25.0)

[warning] File 'ninja-halo-guid-sync.ps1' uses Write-Host. Avoid using Write-Host because it might not work in all hosts, does not work when there is no host, and (prior to PS 5.0) cannot be suppressed, captured, or redirected. Instead, use Write-Output, Write-Verbose, or Write-Information.

(PSAvoidUsingWriteHost)